great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -675,6 +914,17 @@ def _pandas_map_condition_index(
         if filter_column_isnull:
             df = df[df[column_name].notnull()]
 
+    elif "column_list" in accessor_domain_kwargs:
+        column_list = accessor_domain_kwargs["column_list"]
+
+        for column_name in column_list:
+            if column_name not in metrics["table.columns"]:
+                raise ge_exceptions.ExecutionEngineError(
+                    message=f'Error: The column "{column_name}" in BatchData does not exist.'
+                )
+
+        df = df[column_list]
+
```

Comment:
@jcampbell I agree with the direction for moving the keys and the checks as described.  However, I believe that we need this section for the time being because of the following reasons:
- A checkpoint validation includes expectation suites;
- An expectation suite may include a multicolumn expectation
- A multicolumn expectation may be authored without a batch of data and specify a non-existent column in its `column_list` argument.

This is analogous to the single column expectations -- all cases from the invocation coming from upstream in Great Expectations.